### PR TITLE
GEN-831 | Partner Init: add debugger page to initiate partner session

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Create debugger page

- Add form to create a new trial contract

- Implement API route to handle trial contract creation

- Only show debugger page if explicitly enabled (feauture flag)

## Justify why they are needed

- I tried setting it up in App Router but the DX ergonomics were not great

- I do prefer the new API handlers in App Router so I decided to try them out

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2880** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2880" target="_blank">Graphite</a> 
    * **PR #2882** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2882" target="_blank">Graphite</a> 
      * **PR #2883** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2883" target="_blank">Graphite</a> 
        * **PR #2884** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2884" target="_blank">Graphite</a> 
          * **PR #2887** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2887" target="_blank">Graphite</a> 
            * **PR #2890** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2890" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2890?utm_source=stack-comment).

File: apps/store/src/app/api/trials/route.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  const ssn = url.searchParams.get('ssn')
+
+  const apiKey = url.searchParams.get('partner')
+  if (!apiKey) {
+    console.info('No partner API key provided')
+    return NextResponse.redirect(`${ORIGIN_URL}/debugger?success=false`)
+  }
+
+  const body = {
+    ...DEFAULT_BODY,
```
Comment (alebedev) [inline apps/store/src/app/api/trials/route.ts:45]: Do you want to take `requestId` from request?

File: apps/store/src/app/api/trials/route.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  console.debug('Trials debugger | Body = ', body)
+
+  const response = await fetch(ENDPOINT, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'Accept-Language': 'sv-se',
+      Authorization: `Basic ${btoa(`${apiKey}:`)}`,
+    },
+    body: JSON.stringify(body),
```
Comment (alebedev) [inline apps/store/src/app/api/trials/route.ts:62]: Do we need any cache-control settings to make NextJs aware that this request should not be cached?

File: apps/store/src/app/api/trials/route.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  if (!response.ok) {
+    console.info(`Failed to initialize partner widget session: ${response.status}`)
+    console.debug(await response.text())
+    return NextResponse.redirect(`${ORIGIN_URL}/debugger?success=false`)
+  }
+
+  const data = (await response.json()) as ResponseData
+  console.info(`Trials debugger | Widget URL = ${data.widgetUrl}`)
+
+  if (redirectToWidget) {
```
Comment (alebedev) [inline apps/store/src/app/api/trials/route.ts:74]: Nit: if this is the first time we're using `redirectToWidget`, let's declare it here, right before usage

File: apps/store/src/app/api/trials/route.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    console.info('Redirecting to Partner Widget')
+    return NextResponse.redirect(data.widgetUrl)
+  }
+
+  const nextUrl = new URL('/api/partner/init', ORIGIN_URL)
```
Comment (alebedev) [inline apps/store/src/app/api/trials/route.ts:79]: Nit: can you extract this URL building into separate function and comment why we're doing it?

**Review by alebedev — APPROVED** [review]
Looks good, minor comments

File: apps/store/src/app/api/trials/route.ts (hunk 4 above)
Comment (guilhermespopolin) [reply apps/store/src/app/api/trials/route.ts:79]: Yeah, that would be good 🙂

**Review by guilhermespopolin — APPROVED** [review]
I think it looks good.

File: apps/store/src/app/api/trials/route.ts (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/app/api/trials/route.ts:62]: Probably not needed since it is a POST request

**Comment by robinandeer** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2890) rebased this pull request as part of a merge.

**Comment by robinandeer** [comment]
@robinandeer merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2890).

**Pull request merged**